luaui/barwidgets.lua

```diff
@@ -297,49 +296,7 @@ local doMoreYield = (Spring.Yield ~= nil);
 
 local function Yield()
 	if doMoreYield then
-		local doMoreYield = Spring.Yield()
-		if doMoreYield == false then --GetThreadSafety == false
-			--Spring.Echo("WidgetHandler Yield: entering critical section")
-		end
-	end
-end
-
-local function GetWidgetInfo(name, mode)
-
-	do
-		return
-	end -- FIXME
-
-	local lines = VFS.LoadFile(name, mode)
-
-	local infoLines = {}
-
-	for line in lines:gmatch('([^\n]*)\n') do
-		if not line:find('^%s*%-%-') then
-			if line:find('[^\r]') then
-				break -- not commented, not a blank line
-			end
-		end
-		local s, e, source = line:find('^%s*%-%-%>%>(.*)')
-		if source then
-			table.insert(infoLines, source)
-		end
-	end
-
-	local info = {}
-	local chunk, err = loadstring(table.concat(infoLines, '\n'))
-	if not chunk then
-		Spring.Echo('not loading ' .. name .. ': ' .. err)
-	else
-		setfenv(chunk, info)
-		local success, err = pcall(chunk)
-		if not success then
-			Spring.Echo('not loading ' .. name .. ': ' .. err)
-		end
-	end
-
-	for k, v in pairs(info) do
-		Spring.Echo(name, k, 'type: ' .. type(v), '<' .. tostring(v) .. '>')
+		Spring.Yield()
```

Comment:
I'm fairly sure the original code was incorrect and should have been assigning to the global `doMoreYield = Spring.Yield()`

Reply:
After looking at it, I'm thinking the same thing.

I tested it with your suggested modification, and the result was that when loading a typical game (or doing `/luaui reload`), it would call yield only once. Without the modification, it called it once for each widget load (every time `barwidgets.Yield()` was called)

Reply:
(If we do want to change this, I'll move that to another PR as it's a functional change)

Reply:
Looks like yield only even does anything if `LoadingMT=1`, which isn't the case for BAR unless the setting is changed directly.

Some discussion about `Spring.Yield()` that I found: https://discord.com/channels/549281623154229250/724924957074915358/1060147690371039243

---

I'll move modifications of this yield code to another PR.